goodtables/checks/unique_constraint.py

```diff
@@ -11,23 +11,29 @@
 # Module API
 
 @check('unique-constraint', type='schema', context='body')
-def unique_constraint(errors, columns, row_number, state):
-    rindexes = state.setdefault('cache', {})
-    for column in columns:
-        if len(column) == 4:
-            if column['field'].constraints.get('unique'):
-                rindex = rindexes.setdefault(column['number'], {})
-                references = rindex.setdefault(column['value'], [])
-                if references:
-                    # Add error
-                    message = spec['errors']['unique-constraint']['message']
-                    message = message.format(
-                        row_numbers=', '.join(map(str, references + [row_number])),
-                        column_number=column['number'])
-                    errors.append({
-                        'code': 'unique-constraint',
-                        'message': message,
-                        'row-number': row_number,
-                        'column-number': column['number'],
-                    })
-                references.append(row_number)
+class UniqueConstraint(object):
+
+    # Public
+
+    def __init__(self, **options):
+        self.__row_indexes = {}
+
+    def check_row(self, errors, columns, row_number):
+        for column in columns:
+            if len(column) == 4:
```

Comment:
The same as above (#198). It will be a good amount of writing stuff))) Also going to add comments and split on blocks current checks to make it easy to learn from it. So this line should be absolutely unclear for anyone except me for now. I'll document it.

In short `column` could not have `column.value` or `column.header` if it's not healthy column. Check for len is kinda lazy so I'll updated this condition to make it self-explanatory.